### ISSUE TITLE
Support for Python 3.10

### ISSUE THREAD

**Issue by fniephaus** [oracle/graalpython#286]
### TL;DR

We plan to update GraalPy with support for Python 3.10 to allow Python developers to take advantage of the latest Python language features.

### Goal

As of the 22.2 release, GraalPy is limited to Python 3.8. Both the [3.9](https://docs.python.org/3/whatsnew/3.9.html) and [3.10](https://docs.python.org/3/whatsnew/3.10.html) Python releases have introduced many new features such as Structural Pattern Matching, union operators for `dict`, and new string methods. We plan to implement support for these features and upgrade the supported Python language version in GraalPy to 3.10.

### Non-Goal

Keep dedicated support for Python 3.8.

**Comment by tushanfufu** [comment]
When will the Python plugin for Windows be released？
